pylua/test/test_timer.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+import pylua.timer
+import logging
+import time
+import threading
+
+logger = logging.getLogger(__name__)
+
+class TestPyTimer:
+	__test_timer = None
+	__start_timer = None
+	__finish_timer = None
+	__interval = None
+	__cond_var = None
+
+	def timer_task(self):
+		self.__cond_var.acquire()
+		self.__finish_timer = float(time.time())
+		self.__cond_var.notify()
+		self.__cond_var.release()
+		pass
+
+	def __init__(self):
+		self.__interval = 1
+		self.__test_timer = timer.PyTimer(self.timer_task, self.__interval)
+		self.__cond_var = threading.Condition()
+		pass
+
+	def run_cond_var(self, wait_time):
+		self.__cond_var.acquire()
+		self.__cond_var.wait(wait_time)
+		self.__cond_var.release()
+		pass
+
+	def test_run_timer(self, interval = None):
```

Comment:
@malirod I have other test cases: test_stop_timer, test_reset_timer, which use timer with external task, and If I rework task to internal, I have to use internal task for test cases: test_stop_timer, test_reset_timer. And I think  to rework task to internal method is bad way